### ISSUE TITLE
Mypy plugin can't handle mypy development builds

### ISSUE THREAD

**Issue by silasary** [pydantic/pydantic#3883]
### Checks

* [x] I added a descriptive title to this issue
* [x] I have searched (google, github) for similar issues and couldn't find anything
* [x] I have read and followed [the docs](https://pydantic-docs.helpmanual.io/) and still think this is a bug

# Bug

Output of `python -c "import pydantic.utils; print(pydantic.utils.version_info())"`:
```
             pydantic version: 1.9.0
            pydantic compiled: True
                 install path: C:\Users\Clock\.virtualenvs\Penny-Dreadful-Tools-AXkPbsJM\Lib\site-packages\pydantic
               python version: 3.10.2 (tags/v3.10.2:a58ebcc, Jan 17 2022, 14:12:15) [MSC v.1929 64 bit (AMD64)]
                     platform: Windows-10-10.0.19043-SP0
     optional deps. installed: ['dotenv', 'typing-extensions']
```
Mypy Version: `0.940+dev.b3752a60af3f8e6a1b17c4ebc90ed554cb934288`

## Steps to reproduce:
```bash
# Install a development build of mypy
git clone https://github.com/python/mypy.git
cd mypy
python3 -m pip install --upgrade .

# cd to your project, and run mypy
mypy .
```
Output:
> mypy.ini:2:1: error: Error importing plugin "pydantic.mypy": could not convert string to float: '0.940+dev.b3752a60af3f8e6a1b17c4ebc90ed554cb934288'  [misc]
> Found 1 error in 1 file (errors prevented further checking)

**Comment by KotlinIsland** [comment]
https://github.com/samuelcolvin/pydantic/pull/3783
https://github.com/samuelcolvin/pydantic/pull/3845

**Comment by Kartstig** [comment]
I'm seeing the following in my CI:
```
mypy.ini:2: error: Error importing plugin "pydantic.mypy": could not convert string to float: '1.0.1'  [misc]
```
It seems to work fine locally and only encounters this in TeamCity (of course). My mypy.ini lines are as follows

```
[mypy]
plugins = sqlalchemy.ext.mypy.plugin,pydantic.mypy
...
```

Here's the output of `python -c "import pydantic.utils; print(pydantic.utils.version_info())"` on my CI:

```
               pydantic version: 1.9.1
03:55:39               pydantic compiled: True
03:55:39                    install path: /opt/buildagent/work/a6b7174e1ff9fed1/venv/lib/python3.8/site-packages/pydantic
03:55:39                  python version: 3.8.12 (default, Nov 17 2021, 16:58:51)  [GCC 10.2.1 20210110]
03:55:39                        platform: Linux-5.15.0-60-generic-x86_64-with-glibc2.2.5
03:55:39        optional deps. installed: ['dotenv', 'typing-extensions']
```

So essentially the same thing, except this is trying to turn `1.0.1` into a float somewhere? :thinking:

**Comment by samuelcolvin** [comment]
This was fixed at some point, you need to upgrade pydantic, afraid I'm afk and can't remember in which release it was fixed.

**Comment by KotlinIsland** [comment]
Nah, This was resolved in 1.10

**Comment by samuelcolvin** [comment]
Pydantic made a fix some time ago to accept version formats which aren't valid as floats.

**Comment by KotlinIsland** [comment]
Yeah, I did it. #3783

**Comment by samuelcolvin** [comment]
Oh great, sorry I was on my phone and missed that. Your change was included in `v1.10`.

I think that's the problem @Kartstig was having - they're using an older version of pydantic.

**Comment by Kartstig** [comment]
Looks like it's working fine with `1.10.5` but now I have some new errors to resolve :sweat: 

Thanks for the response and clarification!
